### PR TITLE
Add Vial support for NCR-80 Hotswap

### PR DESCRIPTION
**Pull request by juanlufont**
Add Vial keymap for the NCR-80 hotswap version

### CODE REVIEW

File: keyboards/mt/ncr80/hotswap/hotswap.h
Comment (lesshonor) [inline keyboards/mt/ncr80/hotswap/hotswap.h:1]: Please revert the changes to this file. They' aren't necessary.

File: keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
+    KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_INS,   KC_HOME, KC_PGUP,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,    KC_DEL,   KC_END,  KC_PGDN,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,                                KC_UP,
+    KC_LCTL, KC_LALT,                   KC_SPC,                                      KC_RALT,          KC_RCTL,                      KC_LEFT,  KC_DOWN, KC_RGHT),
+  [1] = LAYOUT(
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET,
```
Comment (lesshonor) [inline keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c]:
```suggestion
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, QK_BOOT,
```

File: keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    _______, _______,                   _______,                                     _______,          _______,                       _______, _______, _______),
+  [3] = LAYOUT(
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,     _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                                _______,
+    _______, _______,                   _______,                                     _______,          _______,                       _______, _______, _______),
```
Comment (lesshonor) [inline keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c]:
```suggestion
    _______, _______,                   _______,                                     _______,          _______,                       _______, _______, _______)
```
You don't need all these extra blank layers; they'll automatically be created. There are also too many of them (you only have `#define DYNAMIC_KEYMAP_LAYER_COUNT 3`...)

File: keyboards/mt/ncr80/hotswap/keymaps/vial/rules.mk [hunk 3]
```diff
@@ -0,0 +1,8 @@
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
+LTO_ENABLE = yes
+
+COMBO_ENABLE = no
+KEY_OVERRIDE_ENABLE = no
+QMK_SETTINGS = no
+TAP_DANCE_ENABLE = yes
```
Comment (lesshonor) [inline keyboards/mt/ncr80/hotswap/keymaps/vial/rules.mk]:
```suggestion
```
I'd leave all this enabled IMO (tap dance is enabled by default)

Maybe you want to disable the other things because atmega32u4 doesn't have enough RAM, but `QMK_SETTINGS` is one of Vial's killer features -- so it's a shame to turn it off when you still have so much room left in flash.

**Review by lesshonor — CHANGES_REQUESTED** [review]

File: keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c (hunk 2 above)
Comment (juanlufont) [reply keyboards/mt/ncr80/hotswap/keymaps/vial/keymap.c]: I reused files from another variant of the NCR-80,  hence the extra "baggage". I will change it as suggested.

File: keyboards/mt/ncr80/hotswap/hotswap.h
Comment (juanlufont) [reply keyboards/mt/ncr80/hotswap/hotswap.h:1]: I was not happy with the not-so-perfect indentation of some of the values,  they created a bit of visual noise when checking the details about rows and columns. But if it is required, I will revet these changes.

**Review by lesshonor — APPROVED** [review]
